src/unity/lib/annotation/utils.cpp

```diff
@@ -1,10 +1,79 @@
 #include "utils.hpp"
+#include <logger/assertions.hpp>
+#include <toolkits/image_deep_feature_extractor/image_deep_feature_extractor_toolkit.hpp>
 
-bool turi::annotate::isInteger(std::string s) {
-	if(s.empty() || ((!isdigit(s[0])) && (s[0] != '-') && (s[0] != '+'))) return false;
+namespace turi {
+namespace annotate {
 
-	char * p;
-	strtol(s.c_str(), &p, 10);
+bool is_integer(std::string s) {
+  if (s.empty() || ((!isdigit(s[0])) && (s[0] != '-') && (s[0] != '+')))
+    return false;
 
-	return (*p == 0);
-}
\ No newline at end of file
+  char *p;
+  strtol(s.c_str(), &p, 10);
+
+  return (*p == 0);
+}
+
+gl_sarray
+featurize_images(const std::shared_ptr<turi::gl_sarray> &images) {
```

Comment:
Just use gl_sarray directly -- it can be passed by value or const gl_sarray&

Reply:
done!